code/modules/power/supermatter/supermatter.dm

```diff
@@ -427,25 +428,33 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 		else
 			pluoxiumbonus = 0
 
-		if(h2ocomp >= 0.45 && h2ocomp <= 0.55) //heat protection from h2o only works when it's between 45 and 55%
+		if (h2ocomp >= 0.35 && h2ocomp <= 0.45) //heat protection from h2o only works when it's between 35 and 45%
 			h2obonus = 1
 		else
 			h2obonus = 0
-		if(h2ocomp >= 0.15 && h2ocomp <= 0.55) //the engine will stop producing power from 15% to 55% h2o (slow down the o2 and plasma generation too)
-			h2omalus = 0.1
-		else if (h2ocomp < 0.07)
+
+		if (h2ocomp >= 0.15 && h2ocomp <= 0.45) //the engine will stop producing power from 15% to 45% h2o (slow down the o2 and plasma generation too)
+			h2omalus = 0.05
+		else if (h2ocomp < 0.05)
 			h2omalus = 1
 		else
-			h2omalus = 2  //when gas comp is above 0.55 or between 0.15 and 0.07, the engine will start to freak out (may need test on the amount of freak out)
+			h2omalus = 2  //when gas comp is above 0.45 or between 0.15 and 0.05, the engine will start to freak out (may need test on the amount of freak out)
+
+		if (h2ocomp < 0.30) //variability in the h2o penalty calculation depending on the %
+			h2ofixed = 1
+		else if (h2ocomp >= 0.30 && h2ocomp <= 0.50)
+			h2ofixed = 0.5
+		else
+			h2ofixed = 2
```

Comment:
```suggestion
		if (h2ocomp < 0.30) //variability in the h2o penalty calculation depending on the %
			h2ofixed = 1
		else if (h2ocomp <= 0.50)
			h2ofixed = 0.5
		else
			h2ofixed = 2
```
This should have the same effect

Reply:
Modified, i need to start thinking in different ways to learn to not bloat the code